reminders/models.py

```diff
@@ -30,22 +38,34 @@ class Reminder(models.Model):
         default=ReminderType.WEBSITE,
     )
 
-    def validate_unique_reminder(self, participant, date_time):
-        if Reminder.objects.filter(
-                participant_id__user_id=participant.user_id,
-                participant_id__event_id=participant.event_id,
-                date_time=date_time
-        ):
-            raise ValidationError('reminder already exists')
+    objects = ReminderQuerySet.as_manager()
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(fields=['participant_id', 'date_time'], name='unique reminder'),
+            models.CheckConstraint(check=Q(date_time__gte=Now()), name="date_time__gte_currnet_time")
+        ]
 
     def __str__(self):
         return f"{self.participant_id} - {self.date_time}"
 
-    def clean(self):
-        validate_date(self.date_time)
-        self.validate_unique_reminder(self.participant_id, self.date_time)
-        return super().clean()
-
     def save(self, *args, **kwargs):
-        self.clean()
-        return super().save(*args, **kwargs)
+        time_validation_error = "CHECK constraint failed: date_time__gte_currnet_time"
+        row_duplication_error = ("UNIQUE constraint failed: "
+                                 "reminders_reminder.participant_id_id, "
+                                 "reminders_reminder.date_time")
+
+        try:
+            result = super().save(*args, **kwargs)
+
+        except IntegrityError as error:
+            if row_duplication_error in error.args:
+                raise IntegrityError("reminder already exists")
+
+            elif time_validation_error in error.args:
+                raise IntegrityError("date time should be bigger than the current date_time")
+
+            raise error
+
+        else:
+            return result
```

Comment:
when you have "raise" in the except you don't need the "else" 
```suggestion
        return result
```

Reply:
done. thanks 😄